impl_core/src/main/java/io/opencensus/implcore/stats/MeasureMapImpl.java

```diff
@@ -72,6 +78,20 @@ public void record() {
 
   @Override
   public void record(TagContext tags) {
+    recordInternal(RecordUtils.getTagMap(tags));
```

Comment:
This adds the overhead of copying the TagContext into a Map every time we call record.

Reply:
Not really. `RecordUtils.getTagMap` is already on the critical path of `record`: https://github.com/census-instrumentation/opencensus-java/blob/9c4df3dbe085a70850cc466849d33d90ffa846f4/impl_core/src/main/java/io/opencensus/implcore/stats/MutableViewData.java#L151
This PR only moves it from later in the path to an earlier location.

In addition `RecordUtils.getTagMap` won't copy the TagContext into a Map for the existing APIs. It just extracts the internal Map from the TagContext: https://github.com/census-instrumentation/opencensus-java/blob/9c4df3dbe085a70850cc466849d33d90ffa846f4/impl_core/src/main/java/io/opencensus/implcore/stats/RecordUtils.java#L66
For the new API (`recordWithTags`) we'll have to copy into a Map anyway, since the internal map in TagContext is immutable:
https://github.com/census-instrumentation/opencensus-java/blob/9c4df3dbe085a70850cc466849d33d90ffa846f4/impl_core/src/main/java/io/opencensus/implcore/tags/TagContextImpl.java#L41